pkg/client/client.go

```diff
@@ -0,0 +1,315 @@
+/*
```

Comment:
@kayrus we can add under pkg/openstack/client.go ?

Reply:
No, the whole PR is to split openstack and client packages. If I add it back, the binary sizes will increase again.

Reply:
Ah , got it. Its common client package, I assumed common openstack package  . 
But our ultimate goal is to have common openstack package as well , as there is much redundant code there too. Are you planning to work on that as well?

Reply:
> common openstack package

hm, no. current openstack package is primarily used in occm. other packages don't depend on it.

Reply:
My first step is to work on supporting the `Retry-After` throttling header.